### PR TITLE
Add autocommand event when search wraps around

### PR DESCRIPTION
**Pull request by suy**
This is just early RFC to know if the feature would be accepted. The idea is emitting an autocommand event because then UIs can display something more meaningful than a message on the corner, that I very often just miss which makes me lose a lot of time.

I've just tried adding this to my config:

```
let commands = ['kdialog', '--passivepopup', 'Search Wrapped', '1', '--title', 'Neovim']
autocmd SearchWrapped * call jobstart(commands)
```
That is also of great help to me because sometimes I really miss something BIG when I'm digging something like a giant log file and I'm too focused on the line where the cursor is.

I would like this to be independent of showing the warning message controlled by 'shortmess' (because in a UI which supports the event it might be redundant), but I don't know how do you suggest that.

### CODE REVIEW

**Comment by justinmk** [comment]
Definitely will be merged if it has test coverage. See https://github.com/neovim/neovim/blob/master/test/README.md#executing-tests and let us know if you get stuck.

**Comment by suy** [comment]
It's been a while, but I did not forget this. :) I've fixed the implementation, added docs and tests. Hopefully someone will find time to review and give some guidance if something needs to be changed. I wonder if I should add to the docs an example snippet that one can use to make a message of wrapping more noticeable (e.g. a floating window). I'm happy so far using kdialog, but something portable would be much better.

File: src/nvim/search.c
Comment (gpanders) [inline src/nvim/search.c:1357]: Per the style guide, please use curly braces even for single line if statements.

```suggestion
    if (sia && sia->sa_wrapped) {
      apply_autocmds(EVENT_SEARCHWRAPPED, NULL, NULL, false, NULL);
    }
```

**Comment by gpanders** [comment]
functionaltest timed out, but it passed in tsan and asan so I count that as a pass. openbsd failing on `termxx_spec` (flake)

**Review by gpanders — APPROVED** [review]

**Comment by suy** [comment]
Thank you! Is there something that I should do myself? I don't see if I can just retry the failed builds. Or if I should wait for more reviewers to be able to merge?

**Comment by gpanders** [comment]
The changes look good to me and Justin indicated that he’s happy with this addition, so I’ll merge it. But I wanted to make sure other maintainers had a chance to review if they wanted to.

**Pull request merged**
